feat: include contract; yarn start:docker etc.

dckc opened the pull request:
Goal here is to get all the parts from the simple game dapp that we used in [the Nov 1 office hours demo](https://github.com/Agoric/agoric-sdk/discussions/8489) in this one repo.

See also:
 - [start local chain with docker\-compose · Discussion \#8480](https://github.com/Agoric/agoric-sdk/discussions/8480)

I have all the parts working, but I'm struggling to get them all working at the same time.

Stay tuned for

 - [ ] asciinema recording of a clean build
 - [x] git history clean-up
 - [x] integrate update of docker image that has the walletFactory / smart wallet NFT upgrade (11wf)

```
# prereqs: node, yarn, docker-compose

yarn  # UI deps

yarn build:contract
yarn build:proposal

# local chain using docker-compose

yarn start:docker
yarn docker:logs

# need a few 1000 IST to pay for contract bundle installation

yarn make:help
yarn docker:make balance-q
yarn docker:make mint4k
yarn docker:make balance-q

yarn docker:bash
agd query vstorage children published.priceFeed
```

cc @samsiegart @0xpatrickdev @sufyaankhan  @kbennett2000

dckc committed:
chore(contract): add gameAssetContract from smart-wallet test

dckc committed:
chore: adapt refs around gameAssetContract.js, proposal builder

dckc committed:
chore: borrow makeTracer from @agoric/internal

dckc committed:
build: npm script to run contract proposal builder

 - ignore generated core-eval, bundles

dckc committed:
build(contract): lock packages

dckc committed:
fix: work around @babel/code-frame override

dckc commented:
View the recording at:

 - https://asciinema.org/a/618892

dckc committed:
feat: npm scripts to run local chaing with docker

 - start:docker, docker:logs, docker:bash, docker:make, docker:help
 - agd cheat-sheet in the form of a Makefile

dckc commented:
@samsiegart @0xpatrickdev any chance you can reproduce my results and approve?

There are still some rough edges, but I'm reaching a point of diminishing returns with the polishing.

dckc commented:
p.s. [dc-test-contract](https://github.com/agoric-labs/dapp-join-game/tree/dc-test-contract) 1a653c0 is an attempt to add contract testing with `yarn test` but I hit all sorts of trouble. Evidently `agoric init` and `agoric install` really are necessary; we're further than I thought from using the SDK from npm just by adding dependencies to a project like this: https://github.com/Agoric/agoric-sdk/issues/3857#issuecomment-1802237551

dckc committed:
fixup! feat: npm scripts to run local chaing with docker

0xpatrickdev commented on docker-compose.yml:
```diff
@@ -0,0 +1,17 @@
+  agd:
+    # image publication is WIP
+    # cf. https://github.com/Agoric/agoric-3-proposals/issues/6
+    image: ghcr.io/agoric/agoric-3-proposals:main
```
```suggestion
    image: ghcr.io/agoric/agoric-3-proposals:main
    platform: linux/amd64
```

> "Docker Desktop for Mac with Apple Silicon can automatically emulate amd64 architecture on ARM64. When you try to run an image that is not available for ARM64, Docker Desktop uses Rosetta 2 to emulate an AMD64 CPU."

TIL

0xpatrickdev commented on contract/Makefile:
```diff
@@ -0,0 +1,50 @@
+mint4k:
+	make FUNDS=1000$(ATOM) fund-acct
+	cd /usr/src/agoric-sdk && \
+		yarn --silent agops vaults open --wantMinted 4000 --giveCollateral 1000 >/tmp/want4k.json && \
```
I don't know the `agops` signature off hand, but do we need to specify a denom here? The `agd tx bank ... ` command is missing it

```log
$ docker-compose exec agd make -C /workspace/contract mint4k
make: Entering directory '/workspace/contract'
make FUNDS=1000 fund-acct
make[1]: Entering directory '/workspace/contract'
agd tx bank send validator agoric16xuc6qyl09kxhk9u8hukv7dqmvtg442mmn0jhs 1000 \
  --keyring-backend=test --chain-id=agoriclocal --gas=auto --gas-adjustment=1.2 --yes -b block \
  -o json >,tx.json
Error: invalid decimal coin expression: 1000
```

dckc replied:
`agops vaults open` was written as a quick-and-dirty tool to test vaults, so defaulting **ATOM** was handy. This is no longer considered a feature:

 - https://github.com/Agoric/agoric-sdk/issues/8430

a3a19a7 should address the missing denom

0xpatrickdev reviewed (changes requested):
Only dug in lightly, but things are working for me outside of the 2 comments I made. Exciting! 

https://github.com/agoric-labs/dapp-join-game/pull/2#discussion_r1386931838
https://github.com/agoric-labs/dapp-join-game/pull/2#discussion_r1386938408

samsiegart commented:
- Point of confusion: `yarn build:proposal` spits out a suggested agd command, do I have to run that afterward?
- Ran into the same issue with apple silicon, @0xpatrickdev's suggestion got me past that
- `yarn make:help` output a little confusing:
```
samuelsiegart@Samuels-MacBook-Pro dapp-join-game % yarn make:help

yarn run v1.22.19
$ make -C contract list
make -npq : 2> /dev/null | \
		awk -v RS= -F: '$1 ~ /^[^#%]+$/ { print $1 }'
instance-q
.PHONY
mint4k
fund-acct
vote
balance-q
list
gov-q
✨  Done in 0.09s.
```
- Made it the rest of the way through, saw some IST in my balance and the price feeds showed up on the last step 🎉  (I should've timed it, but I don't think it took more than 10 minutes to set everything up)
- Is this all? Should I try starting the UI and making an offer?

dckc commented:
> Should I try starting the UI and making an offer?

yes, please do, @samsiegart

dckc committed:
fix: mac support

> Docker Desktop for Mac with Apple Silicon can automatically emulate amd64 architecture on ARM64. When you try to run an image that is not available for ARM64, Docker Desktop uses Rosetta 2 to emulate an AMD64 CPU.

Co-authored-by: 0xPatrick <patrick@0xpatrick.dev>

dckc commented:
> Point of confusion: `yarn build:proposal` spits out a suggested `agd` command, do I have to run that afterward?

Either that or something analogous. My current thinking is to have folks use the gov proposal builder UI, so I agree that's mostly a distraction.

> yarn `make:help` output a little confusing

It's supposed to be a list of things you can do with `yarn docker:make`. But yeah... it's clear-as-mud in its present form.

dckc committed:
chore: clarify make:help

dckc committed:
docs: simple mainnet-2 style smart wallet app

dckc commented:
@0xpatrickdev , @samsiegart thanks for the feedback.

I integrated the mac docker thing and clarified `make:help`:

```console
~/projects/simple-game$ yarn make:help
yarn run v1.22.19
$ echo 'USAGE: yarn docker:make TARGET
where TARGET is one of...'; cd contract; make list
USAGE: yarn docker:make TARGET
where TARGET is one of...
instance-q
vote
balance-q
mint4k
gov-q
list
fund-acct
Done in 0.04s.
```

I also updated the [README](https://github.com/agoric-labs/dapp-join-game/tree/dc-add-contract#readme).

PTAL.

dckc committed:
docs: getting started
